nemo/collections/tts/torch/data.py

```diff
@@ -980,3 +1072,319 @@ def __getitem__(self, index):
+
+        self.pad_multiple = pad_multiple
+        self.pitch_normalization = pitch_normalization
+        self.pitch_mean = pitch_mean
+        self.pitch_std = pitch_std
+        self.pitch_conditioning = pitch_conditioning
+        self.speaker_conditioning_type = speaker_conditioning_type
+        self.ssl_content_emb_type = ssl_content_emb_type
+
+        if sup_data_dir is None:
+            sup_data_dir = os.path.join(self.base_data_dir, "sup_data")
+        self.sup_data_dir = sup_data_dir
+
+        if self.pitch_normalization == "speaker_wise":
+            self.speaker_stats = {}
+            if speaker_stats_pitch_fp is None:
+                speaker_stats_pitch_fp = os.path.join(sup_data_dir, "speaker_pitch_stats.json")
+
+            assert os.path.exists(speaker_stats_pitch_fp), "speaker_stats_pitch_fp {} does not exist".format(
+                speaker_stats_pitch_fp
+            )
+
+            with open(speaker_stats_pitch_fp, "r") as f:
+                speaker_stats_raw = json.load(f)
+                for key in speaker_stats_raw:
+                    self.speaker_stats[int(key)] = speaker_stats_raw[key]
+
+    def _get_wav_from_filepath(self, audio_filepath):
+        features = AudioSegment.segment_from_file(
+            audio_filepath, target_sr=self.sample_rate, n_segments=-1, trim=self.trim,
+        )
+        audio_samples = features.samples
+
+        audio, audio_length = torch.tensor(audio_samples), torch.tensor(audio_samples.shape[0]).long()
+
+        # pad audio to a multiple of self.pad_multiple
+        if audio.shape[0] % self.pad_multiple != 0:
+            audio = torch.cat(
+                [audio, torch.zeros(self.pad_multiple - audio.shape[0] % self.pad_multiple, dtype=torch.float)]
+            )
+            audio_length = torch.tensor(audio.shape[0]).long()
+
+        return audio, audio_length
+
+    def get_ssl_features(self, wav_text_id):
+        content_emb_fn = f"{self.ssl_content_emb_type}_content_embedding_{wav_text_id}.pt"
+        speaker_emb_fn = f"speaker_embedding_{wav_text_id}.pt"
+        duration_fn = f"duration_embedding_{wav_text_id}.pt"  # embedding just for namesake
+        content_emb_fp = os.path.join(self.sup_data_dir, content_emb_fn)
+        speaker_emb_fp = os.path.join(self.sup_data_dir, speaker_emb_fn)
+        duration_fp = os.path.join(self.sup_data_dir, duration_fn)
+
+        if os.path.exists(content_emb_fp):
+            content_embedding = torch.load(content_emb_fp)
+        else:
+            raise ValueError(f"Content embedding file {content_emb_fp} does not exist")
+
+        if os.path.exists(speaker_emb_fp):
+            speaker_embedding = torch.load(speaker_emb_fp)
+        else:
+            raise ValueError(f"Speaker embedding file {speaker_emb_fp} does not exist")
+
+        if os.path.exists(duration_fp):
+            duration = torch.load(duration_fp)
+        else:
+            raise ValueError(f"Duration file {duration_fp} does not exist")
```

Comment:
Worth mentioning the preprocessing script in these error messages to point users in the right direction?

Reply:
`done